spec/parsers/arw_parser_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'spec_helper'
+
+describe FormatParser::ARWParser do
+  shared_examples 'likely_match for file' do |filename_with_extension|
+    it "matches '#{filename_with_extension}'" do
+      expect(subject.likely_match?(filename_with_extension)).to be_truthy
+    end
+  end
+
+  shared_examples 'no likely_match for file' do |filename_with_extension|
+    it "does not match '#{filename_with_extension}'" do
+      expect(subject.likely_match?(filename_with_extension)).to be_falsey
+    end
+  end
+
+  describe 'likely_match' do
```

Comment:
I get the idea behind building all these examples programatically to be more extensive. It feels a bit odd to have a "describe" section holding this though.
Maybe this would fit better on a `before` section? or maybe we can cover this with a smaller set of examples but list them one-by-one? What do you think?

Reply:
I personally prefer programatically creating test cases over a long list of manually created test cases especially when I want to quickly create combinations for a simple method like this.

There are fewer strings so there is less chance of a mistake or repeated tests and the loops are fairly self explanatory in my opinion. The outcome of this is human readable so understanding the test outcome is not affected by the loop.

`
bundle exec rspec spec/parsers/arw_parser_spec.rb --format documentation

Randomized with seed 41432

FormatParser::ARWParser
  parses Sony ARW fixtures as arw format file
    is able to parse RAW_SONY_SLTA55V.ARW
    is able to parse RAW_SONY_A900.ARW
    should fail to parse JPG
    should fail to parse CR2
    should fail to parse PNG
    is able to parse RAW_SONY_DSC-RX100M2.ARW
    is able to parse RAW_SONY_A700.ARW
    is able to parse RAW_SONY_NEX7.ARW
    should fail to parse NEF
    is able to parse RAW_SONY_ILCE-7RM2.ARW
    should fail to parse TIFF
    is able to parse RAW_SONY_A100.ARW
  likely_match
    matches 'and.another.Arw'
    matches 'another raw file.ARW'
    matches 'one-more.aRW'
    does not match 'another raw file.tiff'
    matches 'raw_file.ARW'
    matches 'raw_file.arW'
    does not match 'raw_file.tiff'
    matches 'one-more.arw'
    does not match 'raw_file.new'
    matches 'raw_file.aRw'
    matches 'another raw file.arW'
    matches 'one-more.aRw'
    matches 'another raw file.ArW'
    matches 'and.another.aRW'
    matches 'and.another.ArW'
    does not match 'raw_file.cr2'
    does not match 'one-more.jpeg'
    matches 'raw_file.ArW'
    does not match 'and.another.jpeg'
    matches 'one-more.Arw'
    matches 'one-more.ArW'
    matches 'one-more.arW'
    matches 'and.another.arw'
    matches 'one-more.ARw'
    does not match 'raw_file.jpeg'
    does not match 'one-more.new'
    does not match 'and.another.tiff'
    does not match 'another raw file.new'
    matches 'raw_file.Arw'
    matches 'raw_file.ARw'
    matches 'another raw file.ARw'
    does not match 'another raw file.jpeg'
    matches 'and.another.ARw'
    matches 'raw_file.arw'
    matches 'another raw file.Arw'
    matches 'another raw file.arw'
    matches 'one-more.ARW'
    does not match 'and.another.new'
    matches 'and.another.aRw'
    matches 'and.another.ARW'
    matches 'another raw file.aRw'
    does not match 'one-more.cr2'
    does not match 'one-more.tiff'
    does not match 'another raw file.cr2'
    matches 'and.another.arW'
    does not match 'and.another.cr2'
    matches 'raw_file.aRW'
    matches 'another raw file.aRW'
`

Reply:
I think it makes sense, for **this** use-case.

I would only argue whether we need to test all those permutations of valid extensions and filenames, as it feels a bit aside from the actual logic that is tested. If it was me, I would test `anything.` + `(arw|tif|cr2|anything-else)`.

Reply:
Yeah the number of test cases is a be a bit overkill (I copied the initial set of all case permutations of extensions from the aac tests).

As we are using regex I do think there is some worth in at least including some different casing in the extension as well as adding some odd file names (eg including a period) to ensure there are no unexpected outcomes.

eg someone accidentally removing the i from the regex would be caught by including .ARW.